pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1251,14 +1251,26 @@ func (cf *changeFrontier) maybeCheckpointJob(frontierChanged, isBehind bool) err
 	}
 
 	if frontierChanged || updateCheckpoint {
-		cf.js.lastFrontierCheckpoint = timeutil.Now()
 		checkpointStart := timeutil.Now()
 		if err := cf.checkpointJobProgress(
 			cf.frontier.Frontier(), frontierChanged, checkpoint, isBehind,
 		); err != nil {
 			return err
 		}
-		cf.metrics.CheckpointHistNanos.RecordValue(timeutil.Since(checkpointStart).Nanoseconds())
+		cf.metrics.FrontierUpdates.Inc(1)
+
+		checkpointDuration := timeutil.Since(checkpointStart)
+		cf.metrics.CheckpointHistNanos.RecordValue(checkpointDuration.Nanoseconds())
+
+		timeSinceLastCheckpoint := timeutil.Since(cf.js.lastFrontierCheckpoint)
+		cf.js.lastFrontierCheckpoint = timeutil.Now()
+
+		if checkpointDuration > timeSinceLastCheckpoint {
```

Comment:
Can this ever be true?  I think the way the code is currently laid out, checkpointDuration is < timeSinceLastCheckpoint by construction.

- lastFrontierCheckpoint set at t1
- checkpoitnStart set at t2
- checkpointDuration taken at t3 
- timeSinceLastCheckpoint taken at t4